modules/service-networking/main.tf

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_compute_global_address" "global_address" {
+  project       = var.project_id
+  name          = var.address_name
+  purpose       = var.address_purpose
+  address_type  = var.address_type
+  prefix_length = var.address_prefix_length
+  network       = var.network_id
+}
+
+resource "google_service_networking_connection" "default" {
+  network                 = var.network_id
+  service                 = "servicenetworking.googleapis.com"
+  reserved_peering_ranges = [google_compute_global_address.global_address.name]
```

Comment:
@q2w thx for the PR. It is possible to create one service networking with multiple global address ranges. If we run out of IP address we can add more by adding more IP address ranges in same service networking object. You cant create multiple instance of specific service networking which is why `reserved_peering_ranges` is a list of string. Can you plz define global address as an object with multiple global address and attach them all in `reserved_peering_ranges` list.
Also plz make `service` a variable as there are different possible values. Here is an [example](https://github.com/GoogleCloudPlatform/terraform-google-netapp-volumes/blob/main/examples/simple_example/main.tf)

Reply:
Thanks for suggestion! Have included both the recommendations!